agent/internal/task_config.go

```diff
@@ -87,6 +88,19 @@ func NewTaskConfig(workDir string, d *apimodels.DistroView, p *model.Project, t
 		return nil, errors.Errorf("cannot find build variant '%s' for task in project '%s'", t.BuildVariant, t.Project)
 	}
 
+	var taskGroupPtr *model.TaskGroup
+	if t.TaskGroup != "" {
+		taskGroupPtr = p.FindTaskGroup(t.TaskGroup)
+		if taskGroupPtr == nil {
+			return nil, errors.New("programmatic error: task group is nil")
```

Comment:
nit: include the task group name, and maybe make the comment a little more descriptive of why it's bad for the task group to be nil. Something along the lines of `programmatic error: task is part of task group '%s' but no such task group is defined in the project`.